catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/api/EndpointUtility.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.api;
+
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import org.codice.ddf.catalog.ui.query.cql.CqlQueryResponse;
+import org.codice.ddf.catalog.ui.query.cql.CqlRequest;
+
+public interface EndpointUtility {
+
+  CqlQueryResponse executeCqlQuery(CqlRequest cqlRequest)
+      throws UnsupportedQueryException, SourceUnavailableException, FederationException;
```

Comment:
❓ If we want to use the `EndpointUtil` code outside of this bundle, can we split the class up into just the pieces that are needed? For example, a service called `CqlQueries` / `CqlQueriesImpl` separate from `EndpointUtil` so we can break up the dependencies?